File: java/rotational-cipher/src/main/java/RotationalCipher.java
public class RotationalCipher {

  private final int A = (int) 'A';
  private final int Z = (int) 'Z';
  private final int a = (int) 'a';
  private final int z = (int) 'z';
  private final int key;

  public RotationalCipher(int key) {
    this.key = key;
  }

  public String rotate(String string) {
    return string.chars()
        .map(this::rotateChar)
        .mapToObj(c -> Character.toString((char)c))
        .reduce("", (a, b) -> a + b);
  }

  private int rotateChar(int c) {
    int shift = c + key;
    return (c >= a && c <= z) ? shift <= z ? shift : (a + shift - z - 1) : 
           (c >= A && c <= Z) ? shift <= Z ? shift : (A + shift - Z - 1) : 
            c;
  }
}

File: csharp/rotational-cipher/RotationalCipher.cs
using System.Linq;

public static class RotationalCipher
{
  public static string Rotate(string text, int shiftKey) =>
    new string(text.Select(c => RotateChar(c, shiftKey)).ToArray());

  private static char RotateChar(char c, int shiftKey)
  {
    if ('a' <= c && c <= 'z') return (char)('a' + (c - 'a' + shiftKey) % 26);
    if ('A' <= c && c <= 'Z') return (char)('A' + (c - 'A' + shiftKey) % 26);
    return c;
  }
}